File: week8/image.m
I = imread('/home/rguktrkvalley/Pictures/enough.jpeg');
red_channel = I(:, :, 1);
green_channel = I(:, :, 2);
blue_channel = I(:, :, 3);

R_vec = red_channel(:);
G_vec = green_channel(:);
B_vec = blue_channel(:);

csvwrite('r.csv', R_vec);
csvwrite('g.csv', G_vec);
csvwrite('b.csv', B_vec);

%for single file
RGB_data = [R_vec, G_vec, B_vec];
csvwrite('rgb.csv', RGB_data);

File: week8/image.py

import matplotlib.image as mpimg
import csv

I = mpimg.imread('/home/rguktrkvalley/Pictures/enough.jpeg')

red_channel = I[:,:,0]
green_channel = I[:,:,1]
blue_channel = I[:,:,2]


R_vec = red_channel.flatten()
G_vec = green_channel.flatten()
B_vec = blue_channel.flatten()


with open('rpy.csv', 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(R_vec)

with open('gpy.csv', 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(G_vec)

with open('bpy.csv', 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(B_vec)
